Add slash command example to quickstart (#639)
Pycord-Development/pycord: docs/quickstart.rst

Let's name this file ``example_bot.py``. Make sure not to name it ``discord.py`` as that'll conflict
with the library.

> There's a lot going on here, so let's walk you through it step by step:

Reviewer: This didn’t need to be changed imo

Author: I thought it makes more sense, but I'm fine with both.

   sure that we ignore messages from ourselves. We do this by checking if the :attr:`Message.author`
   is the same as the :attr:`Client.user`.
5. Afterwards, we check if the :class:`Message.content` starts with ``'$hello'``. If it does,

>    then we send a message in the channel it was used in with ``'Hello!'``.

Reviewer: Why did you remove stuff from this?

Author: Removed it in order to not direct users towards ext.commands when there is a slash example 2 lines below it.

Reviewer: But ext.commands work fully well with slash commands

Author: It does, but isn't what I would consider the logical next step after completing that example, as this text would imply.

Reviewer: I mean prefixed commands have there upside and iirc discord.Bot is still missing some ext.commands features, I am also planning on deprecating discord.Bot soon and making prefix optional in commands.Bot

Author: At this point we are getting into library design talk, and as I cannot really see into the future of how the library will shape up, I don't think I can just the example to that as of now. All of this can be adjusted as the library further develops :)

Reviewer: Ye I just think this shouldn’t be changed but I would like others opinion

Reviewer 2: Please revert this change

    $ python3 example_bot.py

Now you can try playing around with your basic bot.

A Minimal Bot with Slash Commands

> ---------------

Reviewer: Why is the line shorter than the text?

Author: Fixed

As a continuation, let's create a bot that registers a simple slash command!

It looks something like this:

.. code-block:: python3

    import discord

>     bot = discord.Bot()

Reviewer: Basically what @krittick said.

    @bot.event
    async def on_ready():

>         print(f'We have logged in as {bot.user}')

Reviewer 3: ```suggestion
        print(f"We have logged in as {bot.user}")
```

    @bot.slash_command(name="hello", guild_ids=[your, guild_ids, here])
    async def hello_slash(ctx):
        await ctx.respond("Hello!")

>     bot.run('your token here')

Reviewer 3: ```suggestion
    bot.run("your token here")
```

Let's look at the differences compared to the previous example, step-by-step:

1. The first line remains unchanged.
2. Next, we create an instance of :class:`Bot`. This is different from :class:`Client`, as it supports
   slash command creation and other features, while inheriting all the features of :class:`Client`.
3. We then use the :meth:`Bot.slash_command` decorator to register a new slash command. The first

>    keyword-argument, ``name``, sets the command's name to ``hello``. The ``guild_ids`` attribute contains

Reviewer: This makes name sound like a required argument

Author: I think I'll actually just remove name entirely to simplify this example.

    $ python3 example_bot.py

Now you can try playing around with your basic bot.

A Minimal Bot with Slash Commands
---------------------------------

As a continuation, let's create a bot that registers a simple slash command!

It looks something like this:

.. code-block:: python3

    import discord

    bot = discord.Bot()

    @bot.event
    async def on_ready():
        print(f"We have logged in as {bot.user}")

    @bot.slash_command(guild_ids=[your, guild_ids, here])
    async def hello(ctx):
        await ctx.respond("Hello!")

    bot.run("your token here")

Let's look at the differences compared to the previous example, step-by-step:

1. The first line remains unchanged.

> 2. Next, we create an instance of :class:`Bot`. This is different from :class:`Client`, as it supports
>    slash command creation and other features, while inheriting all the features of :class:`Client`.
> 3. We then use the :meth:`Bot.slash_command` decorator to register a new slash command. 

Reviewer 4: ```suggestion
2. Next, we create an instance of :class:`.Bot`. This is different from :class:`.Client`, as it supports
   slash command creation and other features, while inheriting all the features of :class:`.Client`.
3. We then use the :meth:`.Bot.slash_command` decorator to register a new slash command. 
```
I think these are the correct references.
